notify/receivers.go

```diff
@@ -292,3 +316,189 @@ func ProcessNotifierError(config *GrafanaReceiver, err error) error {
 
 	return err
 }
+
+// GrafanaReceiverTyped represents a parsed and validated APIReceiver
+type GrafanaReceiverTyped struct {
```

Comment:
I think it should be called GrafanaReceiver.. but that name is already taken. Probably it makes sense to rename that to GrafanaNotifierSettings and this to GrafanaReceiverConfiguration

Reply:
Remind me again how is this going to be used that will benefit from generics?

Reply:
That is a way to have Metadata for all notifier configs